src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,83 @@
+import { useContext, useEffect, useState } from 'react';
+import { getPeople } from '../../api';
+import { Loader } from '../Loader';
+import { PeopleList } from '../PeopleList/PeopleList';
+import { PeopleFilters } from '../PeopleFilters/PeopleFilters';
+import { getPreparedPeople } from '../../utils/peopleHelper';
+import { PeopleContext } from '../../contexts/PeopleContext';
+
+export const PeoplePage = () => {
+  const { people, setPeople, visiblePeople } = useContext(PeopleContext);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(currentPeople => {
+        setPeople(getPreparedPeople(currentPeople));
+      })
+      .catch(() => {
+        setIsError(true);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
```

Comment:
From one side this looks correct
But - open the People tab, then Home, then People again

When opening the tab for a second time you have a "flash" of the old people table, then it disappears and loading comes, then new content appears.

It would be perfect to work around this unpleasant UX
For instance - you can check if you have already loaded some people then show the old content, but in some "disabled" mode - to prevent selecting a person, while you have not synchronized with the server. Or you may clear the people when leaving the People tab - but neither solution is simple.

No need to fix this here - just wanted to point you to some real-world problems that may potentially appear with complex interfaces.